pkg/vmcp/auth/factory/outgoing.go

```diff
@@ -0,0 +1,166 @@
+		return nil, err
+	}
+
+	return registry, nil
+}
+
+// registerUnauthenticatedStrategy registers the default unauthenticated strategy.
+func registerUnauthenticatedStrategy(registry auth.OutgoingAuthRegistry) error {
+	unauthStrategy := strategies.NewUnauthenticatedStrategy()
+	if err := registry.RegisterStrategy("unauthenticated", unauthStrategy); err != nil {
+		return fmt.Errorf("failed to register default unauthenticated strategy: %w", err)
+	}
+	return nil
+}
+
+// validateConfig validates the configuration structure.
+func validateConfig(cfg *config.OutgoingAuthConfig) error {
+	if cfg.Default != nil && strings.TrimSpace(cfg.Default.Type) == "" {
+		return fmt.Errorf("default auth strategy type cannot be empty")
+	}
+
+	for backendID, backendCfg := range cfg.Backends {
+		if backendCfg != nil && strings.TrimSpace(backendCfg.Type) == "" {
+			return fmt.Errorf("backend %q has empty auth strategy type", backendID)
+		}
+	}
+
+	return nil
+}
+
+// collectStrategyTypes collects all unique strategy types from configuration.
+func collectStrategyTypes(cfg *config.OutgoingAuthConfig) map[string]struct{} {
+	strategyTypes := make(map[string]struct{})
+
+	// Add default strategy type if present
+	if cfg.Default != nil && cfg.Default.Type != "" {
+		strategyTypes[cfg.Default.Type] = struct{}{}
+	}
+
+	// Add all backend strategy types
+	for _, backendCfg := range cfg.Backends {
+		if backendCfg != nil && backendCfg.Type != "" {
+			strategyTypes[backendCfg.Type] = struct{}{}
+		}
+	}
+
+	return strategyTypes
+}
+
+// registerStrategies instantiates and registers each unique strategy type.
+func registerStrategies(registry auth.OutgoingAuthRegistry, strategyTypes map[string]struct{}) error {
+	for strategyType := range strategyTypes {
+		// Skip "unauthenticated" - already registered
+		if strategyType == "unauthenticated" {
+			continue
+		}
+
+		strategy, err := createStrategy(strategyType)
+		if err != nil {
+			return fmt.Errorf("failed to create strategy %q: %w", strategyType, err)
+		}
+
+		if err := registry.RegisterStrategy(strategyType, strategy); err != nil {
+			return fmt.Errorf("failed to register strategy %q: %w", strategyType, err)
+		}
+	}
+
+	return nil
+}
+
+// createStrategy instantiates a strategy based on its type.
+//
+// Each strategy instance is stateless (except token_exchange which has internal caching).
+// This function validates that the strategy type is not empty and returns an appropriate
+// error for unknown strategy types.
+//
+// Parameters:
+//   - strategyType: The type identifier of the strategy to create
+//
+// Returns:
+//   - auth.Strategy: The instantiated strategy
+//   - error: Any error during strategy creation or validation
+func createStrategy(strategyType string) (auth.Strategy, error) {
+	// Validate strategy type is not empty
+	if strings.TrimSpace(strategyType) == "" {
+		return nil, fmt.Errorf("strategy type cannot be empty")
+	}
+
+	switch strategyType {
+	case "header_injection":
```

Comment:
should this be a constant in the strategies package?

Reply:
I files issue #2466 to improve these